### PR TITLE
database

### PR DESCRIPTION
**Pull request by Alex-Diez**
[Rendered](https://github.com/Alex-Diez/blog/blob/database/Database-as-a-pet-project.md)

### CODE REVIEW

File: Database-as-a-pet-project.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Database as a pet project
+
+We are learning anything by practicing it. As a software developer I gain a lot of experience on my daily basis project. However, sometimes I want to try new things. At such a moment I want to start a new pet project.
```
Comment (Dmitry404) [inline Database-as-a-pet-project.md]: So, how many pet projects have you already started?

File: Database-as-a-pet-project.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+* I definitely use different data structures and algorithms.
+* I learn SQL ... actually how to parse and evaluate SQL queries.
+* It is hard so it is interesting.
+
+And here are some cons:
+* It is hard so I can lose interest :).
+* It is more system programming rather Big Data or Web which are dominant on the market.
+* ... some others that I can't consider because I don't see them :).
+
+But for me the most motivating thing is that you can choose to what extent you want to develop such a project. You may write a simple single-threaded or a complex, concurrent application. You may write easy to test in-memory or a quiet hard to test system with IO (I am not consider writing a project without tests). You may use different libraries of data structures and networking libraries which you and your team wouldn't consider to use on your current project. You even can try to write a distributed database.
```
Comment (Dmitry404) [inline Database-as-a-pet-project.md]:
```
to what extent you want
```
Chot didn't get that phrase.

File: Database-as-a-pet-project.md (hunk 1 above)
Comment (Alex-Diez) [inline Database-as-a-pet-project.md]: you may look at my profile :smile:

Comment (Alex-Diez) [inline Database-as-a-pet-project.md]: And BTW, before pulling my leg you'd better reviewed my [PR into our pet project](https://github.com/CodeInten/http2-ws/pull/10)

**Comment by Alex-Diez** [comment]
Closing this PR. Prefer version from #4

**Pull request closed without merging**